test/gtest/hipblaslt_gemm.cpp

```diff
@@ -0,0 +1,384 @@
+    CheckExceptionsWithSkip<float16, d_mask, e_mask>(miopenDataType_t::miopenHalf);
+};
+TEST_P(HipBLASLtGEMMTestHalf, RunHipBLASLtGEMM)
+{
+    using e_mask = enabled<Gpu::gfx94X, Gpu::gfx90A, Gpu::gfx110X>;
+    using d_mask = disabled<Gpu::gfx103X, Gpu::gfx900, Gpu::gfx906, Gpu::gfx908>;
+    RunGemmDescriptors<float16, d_mask, e_mask>(GetParam(), miopenDataType_t::miopenHalf);
+};
+INSTANTIATE_TEST_SUITE_P(HipBLASLtGEMMTestSet,
+                         HipBLASLtGEMMTestHalf,
+                         testing::ValuesIn(GetTestCases()));
+
+TEST_F(HipBLASLtGEMMTestBFloat16, CheckHipBLASLtGEMMException)
+{
+    using e_mask = enabled<Gpu::gfx94X, Gpu::gfx90A, Gpu::gfx110X>;
+    using d_mask = disabled<Gpu::gfx103X, Gpu::gfx900, Gpu::gfx906, Gpu::gfx908>;
+    CheckExceptionsWithSkip<bfloat16, d_mask, e_mask>(miopenDataType_t::miopenBFloat16);
+};
+TEST_P(HipBLASLtGEMMTestBFloat16, RunHipBLASLtGEMM)
+{
+    using e_mask = enabled<Gpu::gfx94X, Gpu::gfx90A, Gpu::gfx110X>;
+    using d_mask = disabled<Gpu::gfx103X, Gpu::gfx900, Gpu::gfx906, Gpu::gfx908>;
+    RunGemmDescriptors<bfloat16, d_mask, e_mask>(GetParam(), miopenDataType_t::miopenBFloat16);
+};
+INSTANTIATE_TEST_SUITE_P(HipBLASLtGEMMTestSet,
+                         HipBLASLtGEMMTestBFloat16,
+                         testing::ValuesIn(GetTestCases()));
+
+TEST_F(HipBLASLtGEMMTestFloat8, CheckHipBLASLtGEMMException)
+{
+    using e_mask = enabled<Gpu::gfx94X>;
+    using d_mask =
+        disabled<Gpu::gfx103X, Gpu::gfx900, Gpu::gfx906, Gpu::gfx908, Gpu::gfx90A, Gpu::gfx110X>;
+    CheckExceptionsWithSkip<float8, d_mask, e_mask>(miopenDataType_t::miopenFloat8);
+};
+TEST_P(HipBLASLtGEMMTestFloat8, RunHipBLASLtGEMM)
+{
+    using e_mask = enabled<Gpu::gfx94X>;
+    using d_mask =
+        disabled<Gpu::gfx103X, Gpu::gfx900, Gpu::gfx906, Gpu::gfx908, Gpu::gfx90A, Gpu::gfx110X>;
+    RunGemmDescriptors<float8, d_mask, e_mask>(GetParam(), miopenDataType_t::miopenFloat8);
+};
+INSTANTIATE_TEST_SUITE_P(HipBLASLtGEMMTestSet,
+                         HipBLASLtGEMMTestFloat8,
+                         testing::ValuesIn(GetTestCases()));
+
+TEST_F(HipBLASLtGEMMTestBFloat8, CheckHipBLASLtGEMMException)
+{
+    using e_mask = enabled<Gpu::gfx94X>;
+    using d_mask =
+        disabled<Gpu::gfx103X, Gpu::gfx900, Gpu::gfx906, Gpu::gfx908, Gpu::gfx90A, Gpu::gfx110X>;
+    CheckExceptionsWithSkip<bfloat8, d_mask, e_mask>(miopenDataType_t::miopenBFloat8);
+};
+TEST_P(HipBLASLtGEMMTestBFloat8, RunHipBLASLtGEMM)
+{
+    using e_mask = enabled<Gpu::gfx94X>;
+    using d_mask =
+        disabled<Gpu::gfx103X, Gpu::gfx900, Gpu::gfx906, Gpu::gfx908, Gpu::gfx90A, Gpu::gfx110X>;
+    RunGemmDescriptors<bfloat8, d_mask, e_mask>(GetParam(), miopenDataType_t::miopenBFloat8);
+};
+INSTANTIATE_TEST_SUITE_P(HipBLASLtGEMMTestSet,
+                         HipBLASLtGEMMTestBFloat8,
+                         testing::ValuesIn(GetTestCases()));
```

Comment:
Failing to find a heuristic result for FP8 and BF8 in some cases.
Need to figure out this issue.

Reply:
Increasing the m, n, k value fixed the heuristic failing to find for FP8, but still unsure about BF8 failures.

Reply:
BF8 failures are fixed on newer versions of hipBLASLt.

Reply:
Fixed by only enabling BF8 when compiling against newer versions of hipBLASLt